cluster-agent/controllers/argoproj.io/application_controller.go

```diff
@@ -229,3 +382,118 @@ func CompressResourceData(resources []appv1.ResourceStatus) ([]byte, error) {
+		return false
+	}
+	if applicationFromArgoCD.Spec.Destination.Server != applicationFromDB.Spec.Destination.Server {
+		log.Info("Destination.Server field in ArgoCD and DB entry is not in Sync.")
+		log.Info("Destination.Server:= ArgoCD: " + applicationFromArgoCD.Spec.Destination.Server + "; DB: " + applicationFromDB.Spec.Destination.Server)
+		return false
+	}
+	if applicationFromArgoCD.Spec.Destination.Namespace != applicationFromDB.Spec.Destination.Namespace {
+		log.Info("Destination.Namespace field in ArgoCD and DB entry is not in Sync.")
+		log.Info("Destination.Namespace:= ArgoCD: " + applicationFromArgoCD.Spec.Destination.Namespace + "; DB: " + applicationFromDB.Spec.Destination.Namespace)
+		return false
+	}
+	if applicationFromArgoCD.Spec.Destination.Name != applicationFromDB.Spec.Destination.Name {
+		log.Info("Destination.Name field in ArgoCD and DB entry is not in Sync.")
+		log.Info("Destination.Name:= ArgoCD: " + applicationFromArgoCD.Spec.Destination.Name + "; DB: " + applicationFromDB.Spec.Destination.Name)
+		return false
+	}
+	if applicationFromArgoCD.Spec.Project != applicationFromDB.Spec.Project {
+		log.Info("Project field in ArgoCD and DB entry is not in Sync.")
+		log.Info("Project:= ArgoCD: " + applicationFromArgoCD.Spec.Project + "; DB: " + applicationFromDB.Spec.Project)
+		return false
+	}
+	if applicationFromArgoCD.Spec.SyncPolicy.Automated.Prune != applicationFromDB.Spec.SyncPolicy.Automated.Prune {
+		log.Info("Prune field in ArgoCD and DB entry is not in Sync.")
+		return false
+	}
+	if applicationFromArgoCD.Spec.SyncPolicy.Automated.SelfHeal != applicationFromDB.Spec.SyncPolicy.Automated.SelfHeal {
+		log.Info("SelfHeal field in ArgoCD and DB entry is not in Sync.")
+		return false
+	}
+	if applicationFromArgoCD.Spec.SyncPolicy.Automated.AllowEmpty != applicationFromDB.Spec.SyncPolicy.Automated.AllowEmpty {
+		log.Info("AllowEmpty field in ArgoCD and DB entry is not in Sync.")
+		return false
+	}
+	return true
+}
+
+func CleanK8sOperations(ctx context.Context, dbq db.DatabaseQueries, client client.Client, log logr.Logger) {
+	// Get list of Operaions from cluster.
+	listOfK8sOperation := v1alpha1.OperationList{}
+	err := client.List(ctx, &listOfK8sOperation)
+	if err != nil {
+		log.Error(err, "Unable to fetch list of k8s Operation from cluster.")
+		return
+	}
+
+	for _, k8sOperation := range listOfK8sOperation.Items {
+
+		// Skip if Operation was not created by NameSpace Reconciler.
+		if k8sOperation.Annotations["source"] != "periodic-cleanup" {
+			continue
+		}
+
+		// Fetch corresponding DB entry
+		dbOperation := db.Operation{
+			Operation_id: k8sOperation.Spec.OperationID,
+		}
+		if err := dbq.GetOperationById(ctx, &dbOperation); err != nil {
+			log.Error(err, "Unable to fetch DB Operation entry for "+string(k8sOperation.Spec.OperationID))
+			continue
+		}
+
+		if dbOperation.State != "Completed" {
```

Comment:
I think this should be `if dbOperation.State != db.OperationState_Completed || dbOperation.State != db.OperationState_Failed`, feel free to correct my logic if wrong.